src/components/NavigationListener.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { routes } from '../Routes';
+
+const NavigationListener = () => {
```

Comment:
Could this be just a hook? I don't see any benefit in wrapping it in a react component.

Reply:
There is no component that makes sense to have this functionality included.

Reply:
Routes are just routes. 😅 

Creating a wrapper is possible, but still, there would be a component that just does the same, just with children instead of null. Don't see any pros in this approach. 🤔 

Actually, I like this design because it makes the hooks more pluggable without any issues.